shaptools/hana.py

```diff
@@ -447,6 +447,147 @@ def get_sr_status(self):
+                          reconfig=reconfig_option)
+
+        # TODO: return the HANA SQL Statement error if sql fails
+        self._run_hana_command(cmd)
+
+    def unset_ini_parameter(
+            self, database, file_name, layer,
+            section_name, parameter_name, layer_name=None,
+            key_name=None, user_name=None, user_password=None):
+        """
+        unset HANA configuration parameters in ini file
+
+        sql syntax:
+        ALTER SYSTEM ALTER CONFIGURATION (<filename>, <layer>[, <layer_name>])
+        UNSET (<section_name>,<parameter_name>);
+
+        Args:
+            database (str): Database name
+            file_name (str): ini configuration file name
+            layer (str): target layer for the configuration change 'SYSTEM', 'HOST' or 'DATABASE'
+            layer_name (str, optional): target either a tenant name or a target host name
+            section_name (str): section name of parameter in ini file
+            parameter_name (str): name of the parameter to be modified
+            database (str): Database name
+            key_name (str): Key name
+            user_name (str): User
+            user_password (str): User password
+        """
+
+        hdbsql_cmd = self._hdbsql_connect(
+            key_name=key_name, user_name=user_name, user_password=user_password)
+
+        if layer in ('HOST', 'DATABASE') and layer_name is not None:
+            layer_name_str = ", \'" + layer_name + "\'"
+        else:
+            layer_name_str = ''
+
+        cmd = '{hdbsql_cmd} -d {db} '\
+              '\"ALTER SYSTEM ALTER CONFIGURATION (\'{file_name}\', \'{layer}\'{layer_name}) UNSET\
+                  (\'{section_name}\',\'{parameter_name}\');\"'.format(
+                      hdbsql_cmd=hdbsql_cmd, db=database, file_name=file_name, layer=layer,
+                      section_name=section_name, parameter_name=parameter_name,
+                      layer_name=layer_name_str)
+
+        self._run_hana_command(cmd)
+
+    def reduce_memory_resources(
+            self, global_allocation_limit_value, preload_column_tables_value=False,
+            key_name=None, user_name=None, user_password=None):
+        """
+        reduce memory resources needed by hana
+
+        sql example:
+        ALTER SYSTEM ALTER CONFIGURATION ('global.ini', 'SYSTEM') SET
+        ('memorymanager', 'global_allocation_limit') = 'size_in_mb' WITH RECONFIGURE;
+
+        ALTER SYSTEM ALTER CONFIGURATION ('global.ini', 'SYSTEM') SET
+        ('system_replication', 'preload_column_tables') = 'false' WITH RECONFIGURE;
+
+        Args:
+            database (str): Database name
+            key_name (str): Key name
+            user_name (str): User
+            user_password (str): User password
+        """
+        self.set_ini_parameter(database='SYSTEMDB', file_name='global.ini', layer='SYSTEM',\
+        section_name='memorymanager', parameter_name='global_allocation_limit',\
```

Comment:
We should give the option to to use custom data in the parameters. We could do that using kwargs

Reply:
Hi @arbulu89 I have updated the fns to use custom parameters. Basically now this reduce_memory_resources becomes same fn as set_ini_parameter.
As there were 2 specific SQL to reduce the memory resources, I created that method(reduce_memory_resources) with hardcoded SQL parameters. Maybe we no longer need this method?

Reply:
Hi @arbulu89, for the SQL statements to reduce hana resources, just the 'global_allocation_limit' varies. Most of the parameter custom data remains same. For this reason, I earlier created 'reduce_memory_resources' method so I can call the 'set_ini_parameter' fn inside with those hardcoded parameters.

Now while I am writing my state  like 'memory_resources_reduced' in salt-saphana, it needs to fetch/set those parameters (which maybe need to be defined in pillar data or hardcoded in salt-saphana module or here in shaptools). 
Let me know what you think seems good place to define values of these parameters.
For pillar data related to cost-optimized stuff, I am thinking to keep maybe only two parameters like: scenario:cost-optimized and global-allocation-limit:40%

Reply:
Removed the memory_resources fn due to redundancy in functionality with set/unset fns